views/pages/ConfigurationsPage/index.jsx

```diff
@@ -0,0 +1,157 @@
+    try {
+      const configAttributes = {
+        readers: UsersModel.createReadersList(data.readers),
+      }
+      await api.userConfigurations.update(
+        uid,
+        configuration._id,
+        configAttributes
+      )
+
+      await loadAllConfigurations(uid)
+
+      closeForm()
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const loadAllConfigurations = async (userId) => {
+    try {
+      const configurations = await api.userConfigurations.all(userId)
+
+      setConfigurations(configurations)
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const closeForm = () => setConfiguration(null)
+
+  const handleFormModal = async (config) => setConfiguration(config)
+
+  const handleDefaultConfiguration = async (configurationId) => {
+    try {
+      const userAttributes = {
+        ...user,
+        preferences: {
+          ...preferences,
+          config: preferences.config === configurationId ? '' : configurationId,
+        },
+      }
+      const updatedUser = await api.users.update(uid, userAttributes)
+
+      setUser(updatedUser)
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const handleEditConfig = (configurationId) =>
+    navigate(routes.editConfiguration(configurationId))
+  const handleDeleteConfig = async (configurationId) => {
+    try {
+      await api.userConfigurations.destroy(uid, configurationId)
+
+      await loadAllConfigurations(uid)
+
+      if (preferences.config === configurationId)
+        handleDefaultConfiguration(configurationId)
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const handleDuplicateConfiguration = async (configuration) => {
+    try {
+      const { _id, ...configAttributes } = configuration
+      const newConfig = {
+        ...configAttributes,
+        owner: uid,
+        readers: [uid],
+      }
+
+      await api.userConfigurations.create(uid, newConfig)
+      await loadAllConfigurations(uid)
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+
+  return (
+    <Box sx={{ p: '20px' }}>
+      <PageHeader
+        title="Configurations"
+        cta={
+          <Button
+            component={Link}
+            endIcon={<Add />}
+            to={routes.newConfiguration}
+            size="small"
+            sx={{ backgroundColor: 'primary.dark', textTransform: 'none' }}
+            variant="contained"
+          >
+            New Configuration
+          </Button>
+        }
+      />
+      <ConfigurationsTable
+        configurations={configurations}
+        onShare={handleFormModal}
+        onDefaultChange={handleDefaultConfiguration}
+        onEdit={handleEditConfig}
+        onDelete={handleDeleteConfig}
+        onDuplicate={handleDuplicateConfiguration}
+        user={user}
+      />
+      {configuration?._id && formValues.length && (
+        <ShareConfigurationForm
+          formValues={formValues}
```

Comment:
let's use the existing convention of passing the initial values in the same shape as the form expects (`{ readers: [] }`)

```suggestion
          initialValues={initialValues}
```

Reply:
[47a7352](https://github.com/AMP-SCZ/dpdash/pull/487/commits/47a7352aa90256c68d3515ee70b8a513b3052989)